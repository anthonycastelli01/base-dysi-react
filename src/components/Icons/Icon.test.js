import React from 'react'
import TestRenderer from 'react-test-renderer'
import Icon, { Name, Modifier, Backdrop } from './FontAwesome'

describe('components/Icons', () => {
	test('Icon - Basic', () => {
		const name = Name.facebook
		const color = 'red'
		const title = 'Test Title'
		const modifiers = [Modifier.spin]
		const backdrop = Backdrop.square
		// const count = 2;

		const render = TestRenderer.create(
			<Icon
				name={name}
				modifiers={modifiers}
				backdrop={backdrop}
				style={{ color }}
				title={title}
				// count={count}
			/>
		).toJSON()

		// console.log(render.props.dangerouslySetInnerHTML);

		// Snapshot
		expect(render).toMatchSnapshot()

		// Icon
		expect(render.props.dangerouslySetInnerHTML['__html'][0]).toMatch(`data-icon="${Name.facebook}`)

		// Title
		expect(render.props.title).toBe(title)

		// Color
		expect(render.props.style).toMatchObject({ color: color })

		// Spin
		expect(render.props.dangerouslySetInnerHTML['__html'][0]).toMatch('fa-spin')

		// Backdrop
		expect(render.props.dangerouslySetInnerHTML['__html'][0]).toMatch('<mask')

		// Count
		// expect(render.props.dangerouslySetInnerHTML['__html'][0]).toMatch('fa-layers-text');
	})
})
