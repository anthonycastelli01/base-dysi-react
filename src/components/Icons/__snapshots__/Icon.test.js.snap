// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Icons Icon - Basic 1`] = `
<span
  dangerouslySetInnerHTML={
    Object {
      "__html": Array [
        "<svg aria-hidden=\\"true\\" data-prefix=\\"fab\\" data-icon=\\"facebook-f\\" class=\\"svg-inline--fa fa-facebook-f fa-w-14 fa-spin\\" role=\\"img\\" xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 448 512\\"><defs ><clipPath id=\\"clip-2\\"><path fill=\\"currentColor\\" d=\\"M400 32H48C21.5 32 0 53.5 0 80v352c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48V80c0-26.5-21.5-48-48-48z\\"></path></clipPath><mask x=\\"0\\" y=\\"0\\" width=\\"100%\\" height=\\"100%\\" id=\\"mask-1\\" maskUnits=\\"userSpaceOnUse\\" maskContentUnits=\\"userSpaceOnUse\\"><rect x=\\"0\\" y=\\"0\\" width=\\"100%\\" height=\\"100%\\" fill=\\"white\\"></rect><g transform=\\"translate(224 256)\\"><g transform=\\"translate(0, 0)  scale(0.625, 0.625)  rotate(0 0 0)\\"><path fill=\\"black\\" d=\\"M76.7 512V283H0v-91h76.7v-71.7C76.7 42.4 124.3 0 193.8 0c33.3 0 61.9 2.5 70.2 3.6V85h-48.2c-37.8 0-45.1 18-45.1 44.3V192H256l-11.7 91h-73.6v229\\" transform=\\"translate(-132 -256)\\"></path></g></g></mask></defs><rect fill=\\"currentColor\\" clip-path=\\"url(#clip-2)\\" mask=\\"url(#mask-1)\\" x=\\"0\\" y=\\"0\\" width=\\"100%\\" height=\\"100%\\"></rect></svg>",
      ],
    }
  }
  data-radium={true}
  style={
    Object {
      "color": "red",
      "pointerEvents": "none",
    }
  }
  title="Test Title"
/>
`;
